### ISSUE TITLE
Enum keyword recognized as error

### ISSUE THREAD

**Issue by adepke** [MattSutherlin/HLSL_ST3#66]
Scoped enums have been added to DXC, so the following code should be recognized as valid:
```
enum class Hello {
    Test,
    World
};
```
However this highlights the `enum` keyword in red. Below is a screenshot of this happening.
![image](https://user-images.githubusercontent.com/18013792/155036760-44355792-c503-466b-9091-b88981b40774.png)

**Comment by MattSutherlin** [comment]
Agreed.  This has been on my radar for a while and here's my current thinking on it:

I want the plugin to show the right thing for people using DXC without breaking the experience for FXC users.  But supporting DXC means something slightly different depending on which HLSL release we're talking about, too; enum didn't become valid until HLSL 2017.  The plan is that I'd like to extract the bulk of the current syntax into a new base syntax, placing individual leaf syntaxes over it for FXC and the various DXC releases outlined at https://github.com/microsoft/DirectXShaderCompiler/wiki/Language-Versions.  Since probably 90%+ of the language is the same across those versions, if that can all live in the base syntax it makes this all cleaner and easier to maintain (and extend for future versions).  But that also means that individual users need to decide which leaf syntax they want to be using, I don't think it'll be possible for me to figure out automatically.

But that's also a little bit more of a long term plan; I clearly haven't done it yet!  I'll see if there's not something I can do as an acceptable stopgap for now to make this a little less painful for people on the leading edge of the language.  If you've got any ideas I'm all ears, otherwise I'll start tinkering and hopefully get back to you with something useful before too long.

**Comment by adepke** [comment]
Thanks for getting back, I agree that a split between FXC and DXC syntax is a good idea, however I don't see much value creating separate leaf versions for each HLSL release. The main reason for this is that since 2016 each new version is pretty much just additions to the language from the prior release, rather than changing behavior or deprecating syntax. I also think trying to automatically determine this isn't very useful, and instead just setting some editor preference to either FXC or DXC would be fine.

Another thing: the FXC spec is no longer being updated, so to reduce the work needed for supporting new syntax, maybe just make a cut where the plugin is currently for "FXC-mode" and start fresh with HLSL 2021 for "DXC-mode"?
